cmd/notary/keys.go

```diff
@@ -338,7 +365,56 @@ func (k *keyCommander) keysImportRoot(cmd *cobra.Command, args []string) error {
 	}
 	defer importFile.Close()
 
-	err = cs.ImportRootKey(importFile)
+	pemBytes, err := ioutil.ReadAll(importFile)
+	if err != nil {
+		return fmt.Errorf("Error reading input file: %v", err)
+	}
+
+	pemRole := trustmanager.ReadRoleFromPEM(pemBytes)
+
+	// Rewind after reading the first time
+	_, err = importFile.Seek(0, 0)
+	if err != nil {
+		return fmt.Errorf("Error reading input file: %v", err)
+	}
+
+	if pemRole != "" && !data.ValidRole(pemRole) {
+		return fmt.Errorf("Invalid role specified for key: %s", pemRole)
+	}
+
+	if k.keysImportRole != "" && !data.ValidRole(k.keysImportRole) {
+		return fmt.Errorf("Invalid role specified for key: %s", k.keysImportRole)
+	}
+
+	// If the PEM key doesn't have a role in it, we must have --role set
+	if pemRole == "" && k.keysImportRole == "" {
+		return fmt.Errorf("Could not infer role, and no role was specified for key")
+	}
+
+	// If both  PEM role and a --role are provided and they don't match, error
+	if pemRole != "" && k.keysImportRole != "" && pemRole != k.keysImportRole {
+		return fmt.Errorf("Specified role %s does not match role %s in PEM headers", k.keysImportRole, pemRole)
+	}
+
+	// Determine which role to add to between PEM headers and --role flag:
+	var importRole string
+	if k.keysImportRole != "" {
+		importRole = k.keysImportRole
+	} else {
+		importRole = pemRole
+	}
+
+	// If we're importing to targets or snapshot, we need a GUN
+	if (importRole == data.CanonicalTargetsRole || importRole == data.CanonicalSnapshotRole) && k.keysImportGUN == "" {
+		return fmt.Errorf("Must specify GUN for %s key", importRole)
+	}
+
+	cs := cryptoservice.NewCryptoService(k.keysImportGUN, ks...)
+	if importRole == data.CanonicalRootRole {
+		err = cs.ImportRootKey(importFile)
```

Comment:
Even if we don't replace `ImportRootKey` with `ImportRoleKey`, we can probably just call `ImportRoleKey` for root too, so maybe no need for the switch :)